web/src/engine/websites/MangaHot.ts

```diff
@@ -0,0 +1,193 @@
+﻿import { Tags } from '../Tags';
+import icon from './MangaHot.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON, } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+
+type APIResult<T> = {
+    result: null | T
+}
+
+type PageResult = {
+    [id: string]: string
+}
+
+function MangaExtractor(element: HTMLAnchorElement) {
+    return {
+        id: new URL(element.href).searchParams.get('work_code'),
+        title: element.querySelector('div.work_name').textContent.trim()
+    };
+
+}
+
+@Common.MangasSinglePageCSS('/ranking', 'div.ranking a.parent', MangaExtractor)
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangahot', 'マンガほっと (Manga Hot)', 'https://web.mangahot.jp', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/works/detail.php\\?work_code=[^/]+`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const [data] = await FetchCSS(new Request(url), 'div.work_base div.work_name');
+        return new Manga(this, provider, new URL(url).searchParams.get('work_code'), data.textContent.trim());
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const url = new URL(`/works/detail.php`, this.URI);
+        url.search = new URLSearchParams({
+            work_code: manga.Identifier,
+            mode: 'ajax_stories',
+            _: Date.now().toString()
+        }).toString();
+
+        const { result } = await FetchJSON<APIResult<string>>(new Request(url, {
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest',
+            }
+        }));
+        const dom = new DOMParser().parseFromString(result, 'text/html');
```

Comment:
Consider using a more lightweight [approach](https://grrr.tech/posts/create-dom-node-from-html-string/#performance) for creating a sub-tree

Reply:
Hum. Tried it but i fail to see how its able to get nodes from a string.

```
        const dom = document.createDocumentFragment();
        dom.append('result')
```

dont add the proper DOM nodes.

Reply:
Should be quite simple such as
```typescript
const container = document.createElement('template');
container.innerHTML = '<ul><li>a</li><li>b</li></ul>';
const elements = [ ...container.content.querySelectorAll('li') ];
```

Reply:
Done

Reply:
My advise was bad.
The `DOMParser` performs the same or even faster than `innerHTML` and is also more secure to cross scripting attacks.

Reply:
Its the intention that matters boss.